### DISCUSSION TITLE
Opening a .sparsebundle from within a HFS+ volume.

### DISCUSSION THREAD

**Discussion by Sunspark-007** [unsound/hfsexplorer#16] [Q&A] [answered]
Hi, hfsexplorer is able to mount the partition. Inside the partition in the users directory is my .sparsebundle file that will need a password which I have. hfsexplorer does not seem to be able to mount/deal with this folder, it just offers choices like information, extract data, etc. It does not seem to be able to mount/load it as a filesystem if it is already mounting a HFS volume. Does it need to be copied out to a different drive first? Can it be mounted as a filesystem folder without copying it somewhere else first?

**Comment by unsound** [comment]
This is actually a limitation for now. I have done some initial work to enable opening embedded dmgs and sparsebundles directly without having to extract them but it's not yet finished.

**Comment by unsound** [comment]
Hi, I was just looking into this and it turned out that I had done most of the work already so putting the final pieces together was pretty easy. I pushed a commit to the `proposed` branch (9f5aaad) which implements opening images and sparsebundles in a new HFSExplorer window.

**Reply by Sunspark-007** [reply]
Amazing, this is great functionality to have. Thank you! I don't know how to make an .exe from github, so I will wait till a new one is available, but I look forward to testing it.

**Comment by unsound** [comment, marked as answer]
@Sunspark-007 I packaged the latest improvements in a release, version 2021.10.9:
https://github.com/unsound/hfsexplorer/releases/tag/hfsexplorer-2021.10.9

Could you please verify that it solves your problem?

**Comment by Sunspark-007** [comment]
Hi, sorry about the delay--family emergency.

I am very impressed, what you have implemented here worked perfectly!

Process followed: after auto-detecting the device and opening the first partition of the drive, in the sidebar right-clicking on the user.sparsebundle, selecting open in hfsexplorer, upon being presented with a password dialog entered the password. Presented with the choice of a choice of "disk image" (default) or "Apple". When apple partition map is selected, it states can't find handler for partition type "apple_partition_map", selecting disk image opens another hfsexplorer window of the user folder with all folders and files visible. Was able to successfully extract files to my desktop.

Thanks for the great software!
